internal/services/executor/executor.go

```diff
@@ -886,6 +889,8 @@ func (e *Executor) setupTask(ctx context.Context, rt *runningTask) error {
 	}
 
 	_, _ = outf.WriteString("Starting pod.\n")
+	ctx, cancel := context.WithTimeout(ctx, podCreationTimeout)
```

Comment:
If this context is only for NewPod you should assign it to a new variable since it's used also in `e.mkdir`